packages/clerk-js/src/ui/portal/index.tsx

```diff
@@ -24,10 +24,6 @@ export default class Portal<CtxType extends AvailableComponentCtx> extends React
       </ComponentContext.Provider>
     );
 
-    if (componentName === 'UserButton') {
-      return ReactDOM.createPortal(el, node);
-    }
-
```

Comment:
☁️ This change with the addition of a noop Route in UserButton made the removal of `useNavigate` possible.